src/components/SignInForm/index.test.js

```diff
@@ -0,0 +1,148 @@
+        value: "password123",
+      },
+    });
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe("test");
+    expect(screen.getByPlaceholderText(/password/i).value).toBe("password123");
+  });
+
+  it("should display matching error when email domain is invalid", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@test.co",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/password/i), {
+      target: {
+        value: "password123",
+      },
+    });
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@test.co"
+    );
+    expect(screen.getByPlaceholderText(/password/i).value).toBe("password123");
+  });
+
+  it("should display min length error when password is short", async () => {
+    const MIN_CHAR = 8;
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/password/i), {
+      target: {
+        value: `${"a".repeat(MIN_CHAR - 1)}`,
+      },
+    });
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@mail.com"
+    );
+    expect(screen.getByPlaceholderText(/password/i).value).toBe(
+      `${"a".repeat(MIN_CHAR - 1)}`
+    );
+  });
+
+  it("should display max length error when password is long", async () => {
+    const MAX_CHAR = 64;
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/password/i), {
+      target: {
+        value: `${"a".repeat(MAX_CHAR + 1)}`,
+      },
+    });
+
+    fireEvent.submit(screen.getByRole("button"));
+
+    expect(await screen.findAllByRole("alert")).toHaveLength(1);
+    expect(mockSubmit).not.toBeCalled();
+    expect(screen.getByRole("textbox", { name: /email/i }).value).toBe(
+      "test@mail.com"
+    );
+    expect(screen.getByPlaceholderText(/password/i).value).toBe(
+      `${"a".repeat(MAX_CHAR + 1)}`
+    );
+  });
+
+  it("should not display error when value is valid", async () => {
+    fireEvent.input(screen.getByRole("textbox", { name: /email/i }), {
+      target: {
+        value: "test@mail.com",
+      },
+    });
+
+    fireEvent.input(screen.getByPlaceholderText(/password/i), {
+      target: {
+        value: "str0ngp@$$w0rd",
```

Comment:
this should just be "foo" or something, when you put this here the test readers will think this test is somehow related to a password being strong. by making the string useless you are indicating to readers that it's not relevant to the test

Reply:
yeah, you are right, thank you